src/openclsim/core/identifiable.py

```diff
@@ -21,7 +22,20 @@ class Identifiable:
         Note that this field does not influence the simulation.
     """
 
-    def __init__(self, name: str, ID: str = None, *args, **kwargs):
+    def __init__(self, name: str, id: str = None, *args, **kwargs):
+
+        # Deprecation
+        if "ID" in kwargs:
+            if id is not None:
+                raise ValueError("Both ID and id are specified. Use id only.")
+
+            warnings.warn(
+                f"ID argument specified in {self}, please use the attribute id",
+                category=DeprecationWarning,
+                stacklevel=2,
+            )
+            id = kwargs.pop("ID")
```

Comment:
For me, there is no reason the add depreciation warnings I don't think we have such a large user base where this is really necessary.

cool feature though

Reply:
I suggest we keep this until the next major release. For me it also helped to fix all te tests and issues resulting from this change.